### PR TITLE
Update debugging guide and tutorial names

### PR DESCRIPTION
**Pull request by kevinweaver**
## Problem
We have a doc and a guide that sound confusingly similar:
Doc: image
Guide: image
This is somewhat confusing.

## Solution
Judging by the contents of each, I have renamed them:

"Debugging your Contracts" doc -> "Using the Truffle Debugger"
"Debugging a smart contract" guide -> "Debugging an example smart contract"

**NOTE:** - prettier added a lot of additional changes, so I've commented on the important ones for the rename.

### CODE REVIEW

File: partials/docs-sidebar.hbs [hunk 1]
```diff
@@ -14,104 +17,168 @@
...
         </ul>
       {{/each}}
       <h3>Guides</h3>
       <ul>
-        <a href="/guides/debugging-a-smart-contract"><li>Debugging a smart contract</li></a>
-        <a href="/guides/debugger-variable-inspection"><li>Variable Inspection: Going Deeper with the Truffle Solidity Debugger</li></a>
-        <a href="/guides/using-infura-custom-provider"><li>Using Infura (or a custom provider)</li></a>
-        <a href="/guides/deploying-to-the-live-network"><li>Deploying to the Live Network</li></a>
-        <a href="/guides/creating-a-cli-with-truffle-3"><li>Creating an Ethereum-enabled command line tool with Truffle 3.0</li></a>
+        <a href="/guides/debugging-an-example-smart-contract"><li>Debugging a
```
Comment (kevinweaver) [inline partials/docs-sidebar.hbs:92]: Renamed^

File: partials/docs-sidebar.hbs [hunk 2]
```diff
@@ -14,104 +17,168 @@
...
+            <a href="/docs/ganache/{{makeSlug category}}/{{makeSlug this}}"><li
+              >{{this}}</li></a>
           {{/each}}
         </ul>
       {{/each}}
       <h3>Guides</h3>
       <ul>
-        <a href="/guides/debugging-a-smart-contract"><li>Debugging a smart contract</li></a>
-        <a href="/guides/debugger-variable-inspection"><li>Variable Inspection: Going Deeper with the Truffle Solidity Debugger</li></a>
+        <a href="/guides/debugging-an-example-smart-contract"><li>Debugging an
```
Comment (kevinweaver) [inline partials/docs-sidebar.hbs:137]: ^Renamed

File: src/docs/data.json [hunk 3]
```diff
@@ -60,7 +60,7 @@
       "Truffle with MetaMask",
       "Package Management via EthPM",
       "Package Management via NPM",
-      "Debugging Your Contracts",
+      "Using the Truffle Debugger",
```
Comment (kevinweaver) [inline src/docs/data.json:63]: ^Renamed

File: src/docs/truffle/getting-started/using-the-truffle-debugger.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ---
-title: Truffle | Debugging Your Contracts
+title: Truffle | Using the Truffle Debugger
 layout: docs.hbs
 ---
-# Debugging Your Contracts
+# Using the Truffle Debugger
```
Comment (kevinweaver) [inline src/docs/truffle/getting-started/using-the-truffle-debugger.md:5]: ^Renamed

File: src/js/search.js [hunk 5]
```diff
@@ -15,86 +15,79 @@ jQuery(function($) {
         "Interacting with Your Contracts",
         "Package Management via EthPM",
         "Package Management via NPM",
-        "Debugging Your Contracts",
+        "Using the Truffle Debugger",
```
Comment (kevinweaver) [inline src/js/search.js:18]: ^Renamed

File: src/guides/data.json [hunk 6]
```diff
@@ -41,8 +41,8 @@
     "published": true,
     "products": ["truffle", "ganache"]
   },
-  "debugging-a-smart-contract": {
-    "title": "Debugging a smart contract",
+  "debugging-an-example-smart-contract": {
+    "title": "Debugging an example smart contract",
```
Comment (kevinweaver) [inline src/guides/data.json:45]: ^Renamed

File: src/docs/truffle/reference/truffle-commands.md [hunk 7]
```diff
@@ -117,24 +111,21 @@ Interactively debug any transaction on the blockchain.
 truffle debug [<transaction_hash>] [--network <network>] [--fetch-external] [--compile-tests|--compile-all|--compile-none]
 ```
 
-Will start an interactive debugging session on a particular transaction. Allows you to step through each action and replay. See the [Debugging your contracts](/docs/getting_started/debugging) section for more details.
-
+Will start an interactive debugging session on a particular transaction. Allows you to step through each action and replay. See [Using the debugger](/docs/getting_started/using-the-truffle-debugger.html) for more details.
```
Comment (kevinweaver) [inline src/docs/truffle/reference/truffle-commands.md:114]: ^Renamed (the only change that wasn't from prettier in this file.)

File: partials/docs-sidebar.hbs (hunk 1 above)
Comment (fainashalts) [reply partials/docs-sidebar.hbs:92]: Looks like you didn't change the name in the text, or am I missing something?

Comment (fainashalts) [reply partials/docs-sidebar.hbs:92]:
```suggestion
        <a href="/guides/debugging-an-example-smart-contract"><li>Debugging an example
```

**Review by fainashalts — CHANGES_REQUESTED** [review]
Looks good except for one spot I indicated!

File: partials/docs-sidebar.hbs (hunk 1 above)
Comment (kevinweaver) [reply partials/docs-sidebar.hbs:92]: Good catch, thanks!

File: src/docs/truffle/reference/truffle-commands.md [hunk 8]
```diff
@@ -112,24 +106,21 @@ Interactively debug any transaction on the blockchain.
 truffle debug [<transaction_hash>] [--network <network>] [--fetch-external] [--compile-tests|--compile-all|--compile-none]
 ```
 
-Will start an interactive debugging session on a particular transaction. Allows you to step through each action and replay. See the [Debugging your contracts](/docs/getting_started/debugging) section for more details.
-
+Will start an interactive debugging session on a particular transaction. Allows you to step through each action and replay. See [Using the truffle debugger](/docs/getting_started/using-the-truffle-debugger.html) for more details.
```
Comment (kevinweaver) [inline src/docs/truffle/reference/truffle-commands.md:109]: @kevinbluer do i need the .html here?

**Review by cds-amal — CHANGES_REQUESTED** [review]
Hey can you rebase this effort instead of the merge? I'd like the changes to next to each other in the git history. Would be happy to help out!

**Comment by kevinweaver** [comment]
replacing this with #1116 to avoid an ugly merge.

**Pull request closed without merging**